data/meson.build

```diff
@@ -30,25 +30,29 @@ if get_option('daemon')
   )
 endif
 
-con2 = configuration_data()
-con2.set('libexecdir', libexecdir)
-con2.set('bindir', bindir)
-con2.set('localstatedir', localstatedir)
-con2.set('datadir', datadir)
-con2.set('sysconfdir', default_sysconfdir)
+if get_option('systemd')
+  con2 = configuration_data()
+  con2.set('libexecdir', libexecdir)
+  con2.set('bindir', bindir)
+  con2.set('datadir', datadir)
 
-# replace @libexecdir@
-configure_file(
-  input : 'org.freedesktop.fwupd.service.in',
-  output : 'org.freedesktop.fwupd.service',
-  configuration : con2,
-  install: true,
-  install_dir: join_paths(datadir,
-                          'dbus-1',
-                          'system-services'),
-)
+  if systemd.version().version_compare('>= 231')
+    state_dir = join_paths (localstatedir, 'lib', 'fwupd')
+    remotes_dir = join_paths (default_sysconfdir, 'fwupd', 'remotes.d')
+    con2.set('readwritepaths', 'ReadWritePaths=' + state_dir + ' ' + remotes_dir + ' -/boot/efi -/boot -/efi')
```

Comment:
Is there no meson way to do the pythonic `dirs = []`, 'dirs.append()`, `' '.join(dirs)` type thing? This would make it easier to add to the list in the future.